### PR TITLE
[Core/Phases] Implemented phaseID and phasexgroup phase system

### PR DESCRIPTION
**Pull request by SargeroDeV**
This commit does not remove support for phasemask, phase_definitions table support and spell_phase but has been adapted so that they all work in conjunction with the new phaseID implementations for the time being.

### CODE REVIEW

**Comment by jasongdove** [comment]
Awesome to see this! Should this also be removed with this PR?

https://github.com/Legends-of-Azeroth/Legends-of-Azeroth-Pandaria-5.4.8/blob/master/src/server/scripts/Pandaria/WanderingIsland/zone_wandering_island_north.cpp#L110-L152

**Comment by SargeroDeV** [comment]
> Awesome to see this! Should this also be removed with this PR?
> 
> https://github.com/Legends-of-Azeroth/Legends-of-Azeroth-Pandaria-5.4.8/blob/master/src/server/scripts/Pandaria/WanderingIsland/zone_wandering_island_north.cpp#L110-L152

Yes, in this PR I just add those phases to check that they will all work fine, then I will remove that hack. Thanks for the reminder.

File: sql/updates/master/world/20XX_XX_XX_XX_world.sql [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+(40862, 73208, -100),
+(40862, 73212, -100),
+(40863, 76391, -100),
+(40863, 73213, -100),
+(40864, 73210, -100);
+
+SET @OGUID := XXXXXX;
+DELETE FROM gameobject WHERE id IN (210005,210015,210016,210017,210018,210019,210020);
+DELETE FROM `gameobject` WHERE guid BETWEEN @OGUID+0 AND @OGUID+69;
+INSERT INTO `gameobject` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseId`, `phaseGroup`, `position_x`, `position_y`, `position_z`, `orientation`, `rotation0`, `rotation1`, `rotation2`, `rotation3`, `spawntimesecs`, `animprogress`, `state`, `ScriptName`) VALUES
```
Comment (leelf00) [inline sql/updates/master/world/20XX_XX_XX_XX_world.sql:62]: Are these gameobjects from retail sniff data?

**Comment by leelf00** [comment]
@SargeroDeV Do you have some testcase of this PR? How to test it?

**Comment by SargeroDeV** [comment]
> @SargeroDeV Do you have some testcase of this PR? How to test it?

You can create a panda of the class you want and you will see the gameobject that corresponds to your class, or with this last commit I uploaded add quest 29547 and you will see that you will add those 2 phases that add Stormwind and when you reward quest 29548 you will lose phase 1687 and you will stop seeing npc 55789.

**Pull request merged**